src/reducers/fileTree.spec.tsx

```diff
@@ -1065,6 +1073,175 @@ describe(__filename, () => {
     });
   });
 
+  describe('goToRelativeMessage', () => {
+    const _goToRelativeMessage = ({
+      _getRelativeMessage = jest.fn(),
+      _viewVersionFile = jest.fn(),
+      currentMessageUid = '',
+      currentPath = 'file1.js',
+      messageMap = getMessageMap(
+        createFakeExternalLinterResult({ messages: [] }),
+      ),
+      pathList = ['file1.js'],
+      position = RelativePathPosition.next,
+      versionId = 1,
+    } = {}) => {
+      return goToRelativeMessage({
+        _getRelativeMessage,
+        _viewVersionFile,
+        currentMessageUid,
+        currentPath,
+        messageMap,
+        pathList,
+        position,
+        versionId,
+      });
+    };
+
+    it('calls getRelativeMessage', async () => {
+      const _getRelativeMessage = jest.fn();
+      const currentMessageUid = 'message-uid';
+      const currentPath = 'file1.js';
+      const messageMap = getMessageMap(
+        createFakeExternalLinterResult({ messages: [] }),
+      );
+      const pathList = [currentPath];
+      const position = RelativePathPosition.next;
+
+      const { thunk } = thunkTester({
+        createThunk: () =>
+          _goToRelativeMessage({
+            _getRelativeMessage,
+            currentMessageUid,
+            currentPath,
+            messageMap,
+            pathList,
+            position,
+          }),
+      });
+
+      await thunk();
+
+      expect(_getRelativeMessage).toHaveBeenCalledWith({
+        currentMessageUid,
+        currentPath,
+        messageMap,
+        pathList,
+        position,
+      });
+    });
+
+    it('dispatches push with the new location for the message', async () => {
+      const currentPath = 'file1.js';
+      const location = createFakeLocation({ pathname: currentPath });
+      const history = createFakeHistory({ location });
+
+      const line = 1;
+      const uid = 'some-uid';
+      const _getRelativeMessage = jest
+        .fn()
+        .mockReturnValue({ line, path: currentPath, uid });
+      const versionId = 123;
+
+      const { dispatch, thunk } = thunkTester({
+        createThunk: () =>
+          _goToRelativeMessage({
+            _getRelativeMessage,
+            currentPath,
+            versionId,
+          }),
+        store: configureStore({ history }),
+      });
+
+      await thunk();
+
+      expect(dispatch).toHaveBeenCalledWith(
+        push({
+          ...location,
+          search: queryString.stringify({ messageUid: uid, path: currentPath }),
```

Comment:
This needs to use something like [`expect.urlWithTheseParams()`](https://github.com/mozilla/addons-code-manager/blob/d821eeb670e7a24b321654684a4f40fe802ff67a/src/jest-matchers/index.tsx#L32) otherwise the test will rely on the random order of stringifying object keys.